rust/src/ldap/logger.rs

```diff
@@ -0,0 +1,190 @@
+/* Copyright (C) 2024 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+// written by Giuseppe Longo <giuseppe@glongo.it>
+
+use crate::jsonbuilder::{JsonBuilder, JsonError};
+use crate::ldap::filters::*;
+use crate::ldap::ldap::LdapTransaction;
+use crate::ldap::types::*;
+
+fn log_ldap(tx: &LdapTransaction, js: &mut JsonBuilder) -> Result<(), JsonError> {
+    js.open_object("ldap")?;
+
+    if let Some(req) = &tx.request {
+        js.open_object("request")?;
+        js.set_uint("message_id", req.message_id.0 as u64)?;
+
+        match &req.protocol_op {
```

Comment:
Could we have detection keywords as well ?

Reply:
Yes, probably with another PR